src/test_l1_dsAudio.c

```diff
@@ -8219,1674 +7789,600 @@ void test_l1_dsAudio_positive_dsGetAudioCapabilities(void) {
- * |07|Call dsAudioPortTerm() - Terminate audio ports | | dsERR_NONE |  Termination must be successful |
- * |08|Call dsSetSecondaryLanguage() again after terminating audio ports | handle=[valid handle] , sLang=[ENG] | dsERR_NOT_INITIALIZED |call must fail as module is not initialized |
+ * |05|Call dsGetSupportedARCTypes() by looping through valid handles and with a null pointer for type | handle=[valid handle], type=[NULL] | dsERR_INVALID_PARAM | Invalid parameter error must be returned |
+ * |06|Call dsAudioPortTerm() - Terminate audio ports | | dsERR_NONE |  Termination must be successful |
+ * |07|Call dsGetSupportedARCTypes() again after terminating audio ports | handle=[valid handle],type=[valid pointer] | dsERR_NOT_INITIALIZED | dsGetSupportedARCTypes call must fail as module is not intialized |
  *
- * @note Testing dsERR_OPERATION_NOT_SUPPORTED and dsERR_GENERAL might be challenging as they require specific platform conditions
+ * @note Testing dsERR_OPERATION_NOT_SUPPORTED and dsERR_GENERAL might be challenging as they require specific platform conditions.
  */
-void test_l1_dsAudio_negative_dsSetSecondaryLanguage(void) {
-	// Start of the test
-	gTestID = 158; // Assuming Test Case ID is 158
+void test_l1_dsAudio_negative_dsGetSupportedARCTypes(void) {
+	// Logging at the start
+	gTestID = 132;
 	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
-	int result;
-	intptr_t  handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
+	dsError_t result;
+	intptr_t handle[NUM_OF_PORTS];
+	int types[NUM_OF_PORTS];
 
-	// Step 01: Call without initializing audio ports
-	result = dsSetSecondaryLanguage(-1, "ENG");
+	// Step 01: Attempt to get SupportedARCTypes without initializing
+	result = dsGetSupportedARCTypes(-1, &types[0]);
 	CHECK_FOR_EXTENDED_ERROR_CODE( result, dsERR_NOT_INITIALIZED, dsERR_INVALID_PARAM );
-	
+
 	// Step 02: Initialize audio ports
 	result = dsAudioPortInit();
 	UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-	// Step 03: Call with invalid handle
-	result = dsSetSecondaryLanguage(handle[0], "ENG");
+	// Step 03: Attempt to get SupportedARCTypes using an invalid handle
+	result = dsGetSupportedARCTypes(handle[0], &types[0]);
 	UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
 
-	// Step 04: Get a valid port handle
+	// Step 04: Get the port handle for all supported audio ports
 	for (int i = 0; i < NUM_OF_PORTS; i++) {
 		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
 		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
 
-		// Step 05: Call with NULL language pointer
-		result = dsSetSecondaryLanguage(handle[i], NULL);
-		UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
-
-		// Step 06: Call with unsupported language code
-		result = dsSetSecondaryLanguage(handle[i], "XYZ");
-		UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
+		// Step 05: Attempt to get SupportedARCTypes with a null pointer
+			result = dsGetSupportedARCTypes(handle[i], NULL);
```

Comment:
call only for ARC port handle
cosmetic: replace tabs with 4 spaces

Reply:
done